modules/RTC/TPCUtils.js

```diff
@@ -420,16 +449,31 @@ export class TPCUtils {
     }
 
     /**
-    * Enables/disables audio transmission on the peer connection. When
-    * disabled the audio transceiver direction will be set to 'inactive'
-    * which means that no data will be sent nor accepted, but
-    * the connection should be kept alive.
-    * @param {boolean} active - true to enable audio media transmission or
-    * false to disable.
-    * @returns {void}
-    */
-    setAudioTransferActive(active) {
-        this.setMediaTransferActive(MediaType.AUDIO, active);
+     * Resumes or suspends media on the peerconnection by setting the active state on RTCRtpEncodingParameters
+     * associates with all the senders that have a track attached to it.
+     *
+     * @param {boolean} enable - whether media needs to be enabled or suspended.
+     * @returns {Promise} - A promise that is resolved with an array of results when the change is succesful on all the
+     * senders, rejected otherwise.
+     */
+    resumeOrSuspendMedia(enable) {
+        logger.info(`${this.pc} ${enable ? 'Resuming' : 'Suspending'} media transfer.`);
+
+        const senders = this.pc.peerconnection.getSenders().filter(s => Boolean(s.track));
+        const promises = [];
+
+        for (const sender of senders) {
+            const parameters = sender.getParameters();
+
+            if (parameters?.encodings?.length) {
+                for (const encoding of parameters.encodings) {
+                    encoding.active = enable;
+                }
+            }
+            promises.push(sender.setParameters(parameters));
+        }
+
+        return Promise.all(promises);
```

Comment:
Is that what we want? This will resolve with the first failing promise too. I'd say we want `allSettled`.

Reply:
Right , but we do want to resolve with the first failing promise and log an error in the calling function. There is not point in waiting for others to resolve since operation is unsuccessful on one of the senders.

Reply:
Good point! It also means that something has changed in the browser recently and that we need to push all setParameters operations to the jingle session queue. I will try to look up what the spec says and see if some changes have been made in Chrome in that area.